src/cs3panel.tsx

```diff
@@ -89,18 +91,61 @@ export class Cs3HeaderWidget extends ReactWidget {
     }
 }
 
+export const Bottom = (props: BottomProps): JSX.Element => {
+
+    const [text, setText] = useState(props.message);
+
+    const setLabel = async () => {
+        const showHidden: boolean = await props.db.fetch('showHidden') as boolean;
+        if (showHidden == undefined || !showHidden) {
+            const hiddenFilesNo: number = await props.db.fetch('hiddenFilesNo') as number;
+            setText('Hidden files ' + hiddenFilesNo + ' (show)')
+        } else {
+            setText('Cover hidden files')
+        }
+    }
+
+    props.browser.model.pathChanged.connect( async (browser, args) => {
+        await setLabel()
+    })
+
+    props.browser.model.refreshed.connect(async (browser) => {
+        await setLabel()
+    })
+
+    return (
+        <div className={'jp-bottom-div'} onClick={async () => {
+            await setLabel()
+        }}>
+            {text}
+        </div>
+    )
+}
+
 export class Cs3BottomWidget extends ReactWidget {
-    constructor(title: string, id: string, options: Widget.IOptions = {}) {
+    private bottomProps: { message: string; db: IStateDB; drive: CS3Contents, browser: FileBrowser };
+
+    constructor(title: string, id: string, options: Widget.IOptions = {},
+                stateDB: IStateDB,
+                browser: FileBrowser,
+                drive: CS3Contents) {
         super(options);
         this.addClass('c3-bottom-widget');
         this.id = id;
-        this.title.label = title;
-        this.title.caption = title;
         this.title.closable = false;
+        this.bottomProps = {message: 'Show hidden files', db: stateDB, drive: drive, browser: browser}
+        this.node.onclick = async () => {
+            const showHidden = await stateDB.fetch('showHidden')
```

Comment:
'showHidden' is toggled here

Reply:
so why don't we do the `setLabel` here as well? It would be easier to follow as we have a single `onclick` event (I'm assuming that `node` is the `div`, which I'm not sure).

Reply:
But, the fact that you `refresh` the model shouldn't already be changing the text automatically (since you listen for model change)?

Reply:
So this component does not have access to setLabel and it cannot change Bottom component's state, instead it uses onClick to trigger browser's model refreshment, which in turn triggers setting label

Reply:
Btw, would you like to squash commits into single one?

Reply:
Ok, so my second comment made sense. Perfect. I can squash when merging.